log4rs-logstash/src/appender.rs

```diff
@@ -0,0 +1,145 @@
+use anyhow::Result as AnyResult;
+use log::Level as LogLevel;
+use log::Record;
+use log4rs::append::Append;
+use logstash_rs::LogStashRecord;
+use logstash_rs::Sender;
+use logstash_rs::{BufferedSender, TcpSender};
+use std::time::Duration;
+
+pub struct Appender<S>
+where
+    S: Sender,
+{
+    sender: S,
+}
+
+impl<S: Sender> std::fmt::Debug for Appender<S> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}::Appender", module_path!())
+    }
+}
+
+#[derive(Debug)]
+pub struct AppenderBuilder {
+    level: Option<LogLevel>,
+    hostname: String,
+    port: u16,
+    buffer_size: Option<usize>,
+    buffer_lifetime: Option<Duration>,
+    write_timeout: Option<Duration>,
+    connection_timeout: Option<Duration>,
+    use_tls: bool,
+}
+
+impl Default for AppenderBuilder {
+    fn default() -> AppenderBuilder {
+        AppenderBuilder {
+            level: None,
+            hostname: "127.0.0.1".to_string(),
+            port: 5044,
+            buffer_size: Some(1024),
+            buffer_lifetime: Some(Duration::from_secs(1)),
+            write_timeout: Some(Duration::from_secs(10)),
+            connection_timeout: Some(Duration::from_secs(10)),
+            use_tls: false,
+        }
+    }
+}
+
+impl AppenderBuilder {
+    /// Sets threshold for this logger to level.
+    pub fn with_level(&mut self, level: Option<LogLevel>) -> &mut AppenderBuilder {
```

Comment:
I don't think someone will call this with `None` value

Reply:
Below same cases

Reply:
Removed `Option`s